### PR TITLE
Add implementation for legacy customer

### PR DESCRIPTION
**Pull request by rehanjaved237-zz**

### CODE REVIEW

**Comment by stakater-user** [comment]
@rehanjaved237 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/jira-service-desk-operator/commit/fe5d4f55bd8cba5ee9be3e6b6310577edf8ca263/checks) has Failed!

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-58-bd586141`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-58-8aff361d`

**Comment by stakater-user** [comment]
@rehanjaved237 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/jira-service-desk-operator/commit/1395ae1e2adea37ff36b0349b3c61a0eef2b95ee/checks) has Failed!

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-58-7509f5a2`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-58-b8e3c201`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-58-bfc659ba`

File: api/v1alpha1/customer_types.go [hunk 1]
```diff
@@ -117,3 +125,15 @@ func (customer *Customer) IsValidUpdate(existingCustomer Customer) (bool, error)
 
 	return true, nil
 }
+
+func (customer *Customer) IsValidCustomerUpdate(existingCustomer Customer) (bool, error) {
+	if !strings.EqualFold(customer.Spec.Email, existingCustomer.Spec.Email) {
+		// Customer creation takes few seconds to complete. Checking if the customer is Get during the creation process.
+		if existingCustomer.Spec.Name == "User " && existingCustomer.Spec.Email == "?" {
+			return true, nil
+		}
```
Comment (ahmedwaleedmalik) [inline api/v1alpha1/customer_types.go]: It takes a few seconds for customers to be persisted at JSD. Check if it's pending with known values are return.
```suggestion
		// It takes a few seconds for customers to be persisted at JSD. Check if it's pending with known values are return.
		if existingCustomer.Spec.Name == "User " && existingCustomer.Spec.Email == "?" {
			return true, nil
		}
```

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
Please add documentation for `LegacyCustomer` in the readme of the project.

**Review by ahmedwaleedmalik — APPROVED** [review]
LGTM

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-58-209c2a65`

**Pull request merged**